### ISSUE TITLE
inspect.isfunction misses methods bound on objects for context generator

### ISSUE THREAD

**Issue by shuhaowu** [staticjinja/staticjinja#145]
Specifically talking about this line of code: https://github.com/staticjinja/staticjinja/blob/170b027/staticjinja/staticjinja.py#L264

If you have a context generator that's a method on an object, that function call returns False, which causes the `else` block to be executed, which results in the following error:

```
    context.update(context_generator)
TypeError: 'method' object is not iterable
```

Should probably use [`callable(f)`](https://docs.python.org/3/library/functions.html#callable) instead.

**Comment by NickCrews** [comment]
You're right. Toy example:

```
>>> class C:
...     def f(self):
...             return "hi"
... 
>>> c = C()
>>> import  inspect
>>> inspect.isfunction(c.f)
False
```

I'll try to push out a fix today.

**Comment by NickCrews** [comment]
Fixed in those commits linked above, and the fix is released in [version 2.0.0](https://github.com/staticjinja/staticjinja/releases/tag/2.0.0), available on PyPI.

**Comment by NickCrews** [comment]
Thanks @shuhaowu!!
